File: src/models/Story.ts
import {Entity,PrimaryGeneratedColumn,Column,ManyToOne,OneToMany,JoinColumn} from 'typeorm';
import { UserStory } from './UserStory';
import { StoryFilter } from './StoryFilter';
import { StoryRating } from './StoryRating';
import { StoryInfos } from './StoryInfos';
import { StorySuggestion } from './StorySuggestion';

@Entity('story')
export class Story {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'varchar', nullable: true })
  title!: string;

  @Column({ type: 'varchar', nullable: true })
  subtitle!: string;

  @Column({ type: 'text', nullable: true })
  text!: string;

  @Column({ type: 'varchar', nullable: true })
  status!: string; // finalizada, cancelada, em andamento...

  @Column({ type: 'int', nullable: true })
  master_story_id!: number | null;

  @Column({ type: 'varchar', nullable: true })
  language!: string;

  @Column({ type: 'varchar', nullable: true })
  cover!: string;

  @ManyToOne(() => Story, (story) => story.translations, { nullable: true })
  @JoinColumn({ name: 'master_story_id' })
  masterStory!: Story | null;

  @OneToMany(() => Story, (story) => story.masterStory)
  translations!: Story[];

  @OneToMany(() => UserStory, (userStory) => userStory.story)
  userStories!: UserStory[];

  @OneToMany(() => StoryFilter, (filter) => filter.story)
  filters!: StoryFilter[];

  @OneToMany(() => StoryRating, (rating) => rating.story)
  ratings!: StoryRating[];

  @OneToMany(() => StoryInfos, (info) => info.story)
  infos!: StoryInfos[];

  @OneToMany(() => StorySuggestion, (suggestion) => suggestion.story)
  suggestions!: StorySuggestion[];
}

File: src/models/story.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class Story(Base):
    __tablename__ = "story"

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=True)
    subtitle = Column(String, nullable=True)
    text = Column(Text, nullable=True)
    status = Column(String, nullable=True)
    master_story_id = Column(Integer, ForeignKey("story.id"), nullable=True)
    language = Column(String, nullable=True)
    cover = Column(String, nullable=True)

    master_story = relationship(
        "Story",
        back_populates="translations",
        remote_side=[id],
    )
    translations = relationship("Story", back_populates="master_story")

    user_stories = relationship("UserStory", back_populates="story")
    filters = relationship("StoryFilter", back_populates="story")
    ratings = relationship("StoryRating", back_populates="story")
    infos = relationship("StoryInfos", back_populates="story")
    suggestions = relationship("StorySuggestion", back_populates="story")
